singletons-base/singletons-base.cabal

```diff
@@ -1,14 +1,14 @@
 name:           singletons-base
 version:        3.0
-cabal-version:  >= 1.24
+cabal-version:  1.24
 synopsis:       A promoted and singled version of the base library
 homepage:       http://www.github.com/goldfirere/singletons
 category:       Dependent Types
 author:         Richard Eisenberg <rae@cs.brynmawr.edu>, Jan Stolarek <jan.stolarek@p.lodz.pl>
 maintainer:     Ryan Scott <ryan.gl.scott@gmail.com>
 bug-reports:    https://github.com/goldfirere/singletons/issues
 stability:      experimental
-tested-with:    GHC == 8.10.1
+tested-with:    GHC == 8.10.3
```

Comment:
9.0?

Reply:
Morally, this ought to say `tested-with: GHC == 9.0.1`. Unfortunately, I'm not able to do this in practice yet, since `haskell-ci` hasn't caught up to 9.0.1 yet—see my comments in `cabal.haskell-ci`. Until then, I'm temporarily hacking around this restriction with `haskell-ci`'s `github-patches` feature, with the intention that I'll remove this hack once haskell-CI/haskell-ci#474 is resolved. (That issue is about `haskell-ci` lacking 8.10.4 support, but it applies equally to 9.0.1 as well.)

Reply:
I don’t get it. What would go wrong if you simply changed it to `GHC == 9.0.1` by hand?

Reply:
`haskell-ci` uses `tested-with` information to determine which versions of GHC to use when generating a CI configuration file. Moreover, it's very picky about which versions of GHC you can use. If you use a version of GHC that it doesn't know about yet, such as GHC 9.0.1, then it will simply error out when you try to run `haskell-ci regenerate`:

```
$ haskell-ci regenerate
No haskell-ci.sh, skipping bash regeneration
*ERROR* 'tested-with:' specifically refers to unknown 'GHC' versions: 9.0.1
Known GHC versions: 7.0.1, 7.0.2, 7.0.3, 7.0.4, 7.2.1, 7.2.2, 7.4.1, 7.4.2, 7.6.1, 7.6.2, 7.6.3, 7.8.1, 7.8.2, 7.8.3, 7.8.4, 7.10.1, 7.10.2, 7.10.3, 8.0.1, 8.0.2, 8.2.1, 8.2.2, 8.4.1, 8.4.2, 8.4.3, 8.4.4, 8.6.1, 8.6.2, 8.6.3, 8.6.4, 8.6.5, 8.8.1, 8.8.2, 8.8.3, 8.8.4, 8.10.1, 8.10.2, 8.10.3
```

I'd prefer not to have make manual changes each time I update the CI configuration, so I decided to just hack my way around this for now, especially since there are other upcoming changes that depend on this one. I expect this hack to be short-lived, as we can remove it once `haskell-ci` supports GHC 9.0.1 properly.

Reply:
haskell-CI/haskell-ci#474 is now fixed, so I have pushed an update that properly specifies `tested-with: GHC == 9.0.1`.